### PR TITLE
[TEXT-190] Document negative limit for WordUtils abbreviate method

### PR DESCRIPTION
**Pull request by bjmbing**
lower < 0 has no defined behavior in the document. Although it works similar to lower = 0. There is no related document for it.

### CODE REVIEW

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 1]
```diff
@@ -859,7 +859,7 @@ public static boolean isDelimiter(final int codePoint, final char[] delimiters)
     public static String abbreviate(final String str, int lower, int upper, final String appendToEnd) {
         Validate.isTrue(upper >= -1, "upper value cannot be less than -1");
         Validate.isTrue(upper >= lower || upper == -1, "upper value is less than lower value");
-
+        Validate.isTrue(lower >= 0, "lower value cannot be lower than 0");
```
Comment (kinow) [inline src/main/java/org/apache/commons/text/WordUtils.java:862]: @bjmbing thanks for the pull request. Initially this change made sense to me. Then looking at the `abbreviate` code, looks like the `lower` limit is used for `StringUtils.indexOf(str, " ", /* startPos */ lower)`.

`StringUtils.indexOf` javadocs say: "* @param startPos  the start position, negative treated as zero".

Currently, if you pass `-1` or any other negative value, it is treated as zero, in accord with the `StringUtils.indexOf` javadocs.

I think a better solution would be to document that negative values are treated as zero, essentially replicating the `StringUtils` docs. This way, existing code using negative values won't get a runtime exception (which `Validate.isTrue` would do). What do you think?

Comment (bjmbing) [reply src/main/java/org/apache/commons/text/WordUtils.java:862]: Yeah I think the solution you provided is good.

Comment (kinow) [reply src/main/java/org/apache/commons/text/WordUtils.java:862]: Thanks for the prompt reply @bjmbing! If you update this PR with something similar to the javadocs from `StringUtils` for the `lower` limit, we can then create a JIRA issue, and merge this PR :-)

Comment (bjmbing) [reply src/main/java/org/apache/commons/text/WordUtils.java]: Hi @kinow . I have updated the javadoc for lower in WordUtils.abbreviate.

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 2]
```diff
@@ -828,7 +828,7 @@ public static boolean isDelimiter(final int codePoint, final char[] delimiters)
      *
      * @param str         the string to be abbreviated. If null is passed, null is returned.
      *                    If the empty String is passed, the empty string is returned.
-     * @param lower       the lower limit.
+     * @param lower       the lower limit; < -1 value is treated as 0.
```
Comment (kinow) [inline src/main/java/org/apache/commons/text/WordUtils.java:831]: Might be better ", negative treated as zero", as any negative value, not only `-1` is treated as 0

**Review by kinow — CHANGES_REQUESTED** [review]

**Comment by kinow** [comment]
I've created https://issues.apache.org/jira/browse/TEXT-190 for this issue, updated the title. Once the PR passes review, we should be ready to merge and include in the changelog :+1: Thanks @bjmbing !

**Review by kinow — APPROVED** [review]

**Comment by bjmbing** [comment]
I changed the doc as suggested. @kinow . Thank you for your help

**Comment by kinow** [comment]
JIRA issue resolved, added your change with credit/thanks to @bjmbing, merged! Thanks again @bjmbing !

**Pull request closed without merging**